libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -241,6 +267,167 @@ void AP_Camera_Backend::send_camera_settings(mavlink_channel_t chan) const
         NaN);               // focusLevel float, percentage from 0 to 100, NaN if unknown
 }
 
+#if AP_TERRAIN_AVAILABLE
+// calculate poi information
+void AP_Camera_Backend::calculate_poi()
+{
+    while(true) {
+        // run this loop at 10hz
+        hal.scheduler->delay(100);
+        const uint32_t now = AP_HAL::millis();
+        {
+            WITH_SEMAPHORE(poi_calculation.sem);
+            // Do not allow if a specific time is passed after the last request
+            if (now - poi_calculation.req_ms > POI_REQ_TIMOUT_MS) {
+                // This is a stale request
+                continue;
+            }
+        }
+        const AP_AHRS &ahrs = AP::ahrs();
+        Location curr_loc;
+        if (!ahrs.get_location(curr_loc)) {
+            continue;
+        }
+
+        // change vehicle location to ASML
+        curr_loc.change_alt_frame(Location::AltFrame::ABSOLUTE);
+
+        // retrieve gimbal attitude
+        auto *mount = AP_Mount::get_singleton();
+        Vector3f att;
+        if (!mount->get_attitude_euler(get_mount_instance(),att[0], att[1], att[2]) || isnan(att[1]) || isnan(att[2])) {
+            // gimbal attitude unavailable
+            continue;
+        }
+        Quaternion quat;
+        quat.from_euler(att);
```

Comment:
```suggestion
        Quaternion quat;
        quat.from_euler(att);
        if (!mount->get_attitude_quaternion(quat) {
            // gimbal attitude unavailable
            continue;
        }
```

... and create that method on the backend